pkg/oracle/oracle.go

```diff
@@ -148,10 +154,28 @@ func (o *Oracle) Run() {
 	for i := 0; i < o.MainCfg.MaxConcurrentRequests; i++ {
 		go o.runRequestWorker()
 	}
+
+	tick := time.NewTicker(o.MainCfg.MaxTaskTimeout)
```

Comment:
I guess you need to trigger it more often than `MaxTaskTimeout` to get closer to `MaxTaskTimeout` for real timeouts. Also, this ticker can be stopped if there are no requests at the moment.